e2e/test-component/scenarios/embedding-sdk/static-dashboard.cy.spec.tsx

```diff
@@ -79,4 +79,63 @@ describeEE("scenarios > embedding-sdk > static-dashboard", () => {
       cy.icon("expand").should("be.visible"); // enter full screen control
     });
   });
+
+  describe("loading behavior for both entity IDs and number IDs (metabase#49581)", () => {
+    const successTestCases = [
+      {
+        name: "correct entity ID",
+        dashboardIdAlias: "@dashboardEntityId",
+      },
+      {
+        name: "correct number ID",
+        dashboardIdAlias: "@dashboardId",
+      },
+    ];
+
+    const failureTestCases = [
+      {
+        name: "wrong entity ID",
+        dashboardId: "VFCGVYPVtLzCtt4teeoW4",
+      },
+      {
+        name: "one too many entity ID character",
+        dashboardId: "VFCGVYPVtLzCtt4teeoW49",
+      },
+      {
+        name: "wrong number ID",
+        dashboardId: 9999,
+      },
+    ];
+
+    successTestCases.forEach(({ name, dashboardIdAlias }) => {
+      it(`should load dashboard content for ${name}`, () => {
+        cy.get(dashboardIdAlias).then(dashboardId => {
+          mountSdkContent(<StaticDashboard dashboardId={dashboardId} />);
```

Comment:
I wonder if we can share the same test suite for static and interactive dashboard tests (since the only difference is the mounting `StaticDashboard` and `InteractiveDashboard` component), or at least share the test cases.

Reply:
so what helper do you have in mind?

`mountSdkContent(<StaticDashboard dashboardId={dashboardId} />);` is already pretty light imo. The rest are assertions which shouldn't be extracted as the assertion logic should be in the test suites.

Also, these suite aren't quite the same, only the new set of tests are almost the same (the tests for this PR). These suites (static dashboard, and interactive dashboard) already exist and they have different set of tests.

Reply:
@WiNloSt I was thinking of sharing `failureTestCases` and `successTestCases` across files, but I think in this case keeping the assertions as-is would indeed be better. Ignore my comment then 🙏🏻